rapid7_insight_agent/icon_rapid7_insight_agent/connection/connection.py

```diff
@@ -0,0 +1,55 @@
+import insightconnect_plugin_runtime
+from .schema import ConnectionSchema, Input
+# Custom imports below
+from insightconnect_plugin_runtime.exceptions import PluginException, ConnectionTestException
+from icon_rapid7_insight_agent.util.region_map import region_map
+import requests
+
+
+class Connection(insightconnect_plugin_runtime.Connection):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(input=ConnectionSchema())
+
+    def connect(self, params):
+        self.logger.info("Connect: Connecting...")
+
+        # Setup region
+        self.api_key = params.get(Input.API_KEY).get("secretKey")
+        self.org_key = params.get(Input.ORG_ID).get("secretKey")
+        region_string = params.get(Input.REGION)
+        region_code = region_map.get(region_string)
+
+        if region_code:
+            self.endpoint = f"https://{region_code}.api.insight.rapid7.com/graphql"
+        else:
+            # It's an enum, hopefully this never happens.
+            raise PluginException(cause="Region not found.",
+                                  assistance="Region code was not found for selected region. Please contact support.")
+
+        self.logger.info("Setup Complete")
+
+    def get_headers(self):
```

Comment:
This should be moved into an API class

Reply:
I disagree with this...getting headers is really all the connection class does in this instance.

Reply:
The headers should be created once during the initialization of an API object - you won't even need a get_headers function anymore. The init of the class takes an `api_key`, the requests session is created with the headers, and then you no longer need to worry about headers or having a separate function for it.